File: pubnub-gson/src/main/java/com/pubnub/api/endpoints/FetchMessages.java
package com.pubnub.api.endpoints;

import com.pubnub.api.models.consumer.history.PNFetchMessagesResult;

public interface FetchMessages extends Endpoint<PNFetchMessagesResult> {
    FetchMessages channels(java.util.List<String> channels);

    FetchMessages maximumPerChannel(Integer maximumPerChannel);

    FetchMessages start(Long start);

    FetchMessages end(Long end);

    FetchMessages includeMeta(boolean includeMeta);

    FetchMessages includeMessageActions(boolean includeMessageActions);

    FetchMessages includeMessageType(boolean includeMessageType);

    FetchMessages includeUUID(boolean includeUUID);
}

File: pubnub-kotlin/pubnub-kotlin-api/src/commonMain/kotlin/com/pubnub/kmp/endpoints/FetchMessages.kt
package com.pubnub.kmp.endpoints

import com.pubnub.api.models.consumer.history.PNFetchMessagesResult
import com.pubnub.kmp.PNFuture

/**
 * @see [PubNub.fetchMessages]
 */
interface FetchMessages : PNFuture<PNFetchMessagesResult>